vendor/bat-native-ledger/src/bat/ledger/internal/bitflyer/bitflyer_authorization.cc

```diff
@@ -183,31 +186,21 @@ void BitflyerAuthorization::OnClaimWallet(
     return callback(type::Result::LEDGER_ERROR, {});
   }
 
-  const std::string event_text = std::string(constant::kWalletBitflyer) +
-                                 (!wallet_ptr->address.empty() ? "/" : "") +
-                                 wallet_ptr->address.substr(0, 5);
+  DCHECK(!address.empty());
 
-  if (result == type::Result::DEVICE_LIMIT_REACHED) {
-    BLOG(0, "Wallet linking limit reached!");
-
-    ledger_->database()->SaveEventLog(log::kDeviceLimitReached, event_text);
-
-    return callback(type::Result::DEVICE_LIMIT_REACHED, {});
-  }
-
-  if (result == type::Result::MISMATCHED_PROVIDER_ACCOUNTS) {
-    BLOG(0, "Mismatched bitFlyer accounts!");
-
-    ledger_->database()->SaveEventLog(log::kMismatchedProviderAccounts,
-                                      event_text);
-
-    return callback(type::Result::MISMATCHED_PROVIDER_ACCOUNTS, {});
-  }
-
-  if (result != type::Result::LEDGER_OK) {
-    BLOG(0, "Couldn't claim wallet");
-    callback(type::Result::LEDGER_ERROR, {});
-    return;
+  switch (result) {
+    case type::Result::DEVICE_LIMIT_REACHED:
+    case type::Result::MISMATCHED_PROVIDER_ACCOUNTS:
+    case type::Result::REQUEST_SIGNATURE_VERIFICATION_FAILURE:
+      ledger_->database()->SaveEventLog(
+          log::GetEventLogKeyForLinkingResult(result),
+          constant::kWalletBitflyer + std::string("/") + address.substr(0, 5));
+      return callback(result, {});
+    default:
+      if (result != type::Result::LEDGER_OK) {
+        BLOG(0, "Couldn't claim wallet!");
+        return callback(type::Result::LEDGER_ERROR, {});
```

Comment:
Do we want to pass back `result` here or we don't care about the specifics for how this failed?

Reply:
Yeah, good point.
Currently, it shouldn't matter, since based on https://github.com/brave/brave-core/blob/051d47a723de3c33d6827c6d89fd6f229f674caa/vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_bitflyer/post_claim_bitflyer.cc#L61-L96 and
https://github.com/brave/brave-core/blob/051d47a723de3c33d6827c6d89fd6f229f674caa/vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_bitflyer/post_claim_bitflyer.cc#L98-L123 if `result` here is not `LEDGER_ERROR` (or `LEDGER_OK`), then it can only be `NOT_FOUND`, which I don't think would make too much sense, since non-KYC'd users should have already gotten rejected here:
https://github.com/brave/brave-core/blob/051d47a723de3c33d6827c6d89fd6f229f674caa/vendor/bat-native-ledger/src/bat/ledger/internal/bitflyer/bitflyer_authorization.cc#L69-L74 So yeah, we can return `result` here if you want, it shouldn't change behavior. Same applies to Gemini.

Reply:
OK, whatever you think is best is fine by me.

Reply:
Fixed.